### ISSUE TITLE
Why simple QAT is not applicable on RepVGG

### ISSUE THREAD

**Issue by Janus-Shiau** [meituan/YOLOv6#816]
### Before Asking

- [X] I have read the [README](https://github.com/meituan/YOLOv6/blob/main/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [X] I want to train my custom dataset, and I have read the [tutorials for training your custom data](https://github.com/meituan/YOLOv6/blob/main/docs/Train_custom_data.md) carefully and organize my dataset correctly; (FYI: We recommand you to apply the config files of xx_finetune.py.) 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。（FYI: 我们推荐使用xx_finetune.py等配置文件训练自定义数据集。）

- [X] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [X] I have searched the YOLOv6 [issues](https://github.com/meituan/YOLOv6/issues) and found no similar questions.

### Question

I have recently been reading several reparameterization papers, including RepVGG, RepOptimizer, and QARapVGG. While reviewing the RepOptimizer paper, I came across Appendix C, which states that re-parameterized RepVGG models do not have Batch Normalization (BN) layers, and the paper also notes that fine-tuning with Quantization-Aware Training (QAT) can be difficult to implement in this context.

> Even worse, as a re-parameterized RepVGG has no BN layers anymore (and it is tricky to add BN layers into it again without harming the performance), finetuning with a Quantization-Aware Training (QAT) method is not straightforward to implement.

However, I am confused about this issue because the [original paper on QAT](https://arxiv.org/abs/1712.05877) suggests that standard convolutions without BN are acceptable for the quantization process. Therefore, I am unsure why RepVGG QAT implementation is not straightforward after simplifying the inference architecture. Or will it harm the performance?

I would be grateful if you could provide some insight into why adding BN layers back into a re-parameterized RepVGG model can be difficult without harming performance, and why QAT implementation may be challenging in this context. I apologize if my question is not directly related to YOLOv6, and I appreciate any guidance or resources you can provide.

Thank you for your time and expertise.

**Comment by xingyueye** [comment]
@Janus-Shiau In fact, the deploy mode RepVGG could use QAT directly, but the results are not satisfactory because this network does not have BN layers, which makes it difficult to optimize. It is a feasible method to re-insert and re-calib BN layers into deploy model, as given in RepVGG repos. However, it is still not a perfect choice because it introduces additional steps.
The said that `QAT method is not straightforward to implement` mainly because PyTorch's built-in quantization tools may not be good at handling such a models.

**Comment by Janus-Shiau** [comment]
Thank you for your reply. Your explanation is clear and really help me a lot!
